### ISSUE TITLE
Bug: SAM-CLI 1.61.0 is breaking our sam delete with "Max attempts exceeded" error

### ISSUE THREAD

**Issue by hung-tang** [aws/aws-sam-cli#4361]
### Description:

The `sam delete` command has broken our pipeline with the release of 1.61.0.  I suspect it's due to this change [here](https://github.com/aws/aws-sam-cli/pull/4321).  After I rolled back to 1.60.0, our pipeline is working again.

We have Lambdas with VPC connectivity configured.  As documented [here](https://docs.aws.amazon.com/lambda/latest/dg/foundation-networking.html), this can take up to 20 minutes for Lambdas to clean up HENIs.  The latest 1.61.0 release looks like it will timeout prematurely which will return an error to our pipeline job.  Prior to 1.61.0, the `sam delete` command will just wait until stack is deleted (i.e. the Lambdas is able to delete the HENIs)

Please consider rolling back [4321](https://github.com/aws/aws-sam-cli/pull/4321).

### Steps to reproduce:

Make sure you have SAM CLI 1.61.0 installed
1. Create and deploy lambda with VPC connectivity using `sam deploy`
2. Then run `sam delete`

### Observed result:

After a minute, you will see the "Max attempts exceeded" error message:

```
2022-10-31 15:56:13,029 | Using config file: samconfig.toml, config environment: default
2022-10-31 15:56:13,029 | Expand command line arguments to:
2022-10-31 15:56:13,029 | --stack_name=htang-dev2 --no_prompts 
2022-10-31 15:56:13,120 | Config file location: /home/htang/samconfig.toml
2022-10-31 15:56:13,120 | Config file '/home/htang/samconfig.toml' does not exist
2022-10-31 15:56:13,539 | Input stack is deployed, continue deleting
2022-10-31 15:56:13,765 | Stack with id htang-dev2-d1ad6991-CompanionStack does not exist
2022-10-31 15:56:14,102 | Could not find the S3 file with the key 93c23669f53871dd23571c90926e562e
2022-10-31 15:56:14,103 | 	- Could not find and delete the S3 object with the key 93c23669f53871dd23571c90926e562e
2022-10-31 15:56:14,175 | Could not find the S3 file with the key 54250197b1767e869139074ec10db986
2022-10-31 15:56:14,175 | 	- Could not find and delete the S3 object with the key 54250197b1767e869139074ec10db986
2022-10-31 15:56:14,252 | Could not find the S3 file with the key 386858479ff0c5a15b327888dd6a7aa6.template
2022-10-31 15:56:14,252 | 	- Could not find and delete the S3 object with the key 386858479ff0c5a15b327888dd6a7aa6.template
	- Deleting Cloudformation stack htang-dev2
Error: Failed to delete the stack: teststack, msg: ex: Waiter StackDeleteComplete failed: Max attempts exceeded
```

### Expected result:

In <= 1.60.0 releases, the error message above will not occur and the `sam delete` will return successful in about 20 minutes or so.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Linux
2. `sam --version`: 1.61.0-
5. AWS region: us-east-2

`Add --debug flag to command you are running`

**Comment by sriram-mv** [comment]
Thank you for raising this issue, the regression has been addressed on https://github.com/aws/aws-sam-cli/pull/4362. Will add a comment here on when this is targeted to go out.

**Comment by mndeveci** [comment]
Fix is been released with [SAM CLI v1.62.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.62.0)
